### PR TITLE
Enhance JDBC upgrade framework to support multiple upgrade versions for the same test

### PR DESCRIPTION
**Pull request by rohit01010**
### Description
Currently version upgrade test framework does not have support to store different expected output for different upgrade path, to add this support I had restructured the upgrade test script folder as follows:

upgrade
----13_4
--------preparation
--------verification_cleanup
---------schedule
----13_5
--------preparation
--------verification_cleanup 
--------schedule               
----13_6
--------preparation
--------verification_cleanup
------------13_4  
--------schedule
----14_3
--------preparation
--------verification_cleanup
------------13_5
------------13_6
--------schedule
                   
Now all the tests scripts that need to be executed after upgrade should be mentioned in the schedule file of the respective upgrade path folder which have same expected output across different upgrade paths, other test scripts which do not have same expected output across different upgrade path can be added in the preparation and verification_cleanup folder.

Expected output for a test-script in 13_4/preparation folder should be named as 13_4__preparation__<test_filename>-vu-prepare.<file_extension>
Expected output for a test-script in 13_4/verification_cleanup/13_6 folder should be named as 13_4__verification_cleanup__13_6__<test_filename>-vu-verify.<file_extension>  or 13_4__verification_cleanup__13_6__<test_filename>-vu-cleanup.<file_extension> 

Made necessary changes in version upgrade test framework and JDBC test framework to support this restructured upgrade test folder format.

In the input folder in JDBC, replaced each of the test-scripts inside the datatypes folder with it's prepare, verify and cleanup scripts. Also added the expected output corresponding those scripts and removed the expected output of the testscripts I had removed.

Signed-off-by: Rohit Bhagat <rohitbgt@amazon.com>

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/configuration/upgrade-test-configuration.yml [hunk 1]
```diff
@@ -7,4 +7,14 @@ upgrade-version: [[
     version: 14.3,
     upgrade-type: major 
   },
+],
```
Comment (surendravishnoi) [inline .github/configuration/upgrade-test-configuration.yml:10]: let us add one more path from 13.5 - 13.6 - 14.3

File: test/JDBC/upgrade/13_4/schedule [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+TestDatetime2
+TestSmallMoney
+TestDatetime
+TestSQLVariant
+TestDate
+TestText
+
+# Latest version of these files give protocol violation in 13_4 to 13_6 upgrade
+# since fix is in 13_7 so running their modified version for this upgrade path
+# TestDecimal
```
Comment (surendravishnoi) [inline test/JDBC/upgrade/13_4/schedule]: let us remove these from schedule file, these are part of prepare/verify and self explanatory

File: test/JDBC/upgrade/13_4/schedule [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+TestVarChar
+TestXML
+BABEL-3112
+BABEL-3166
+BABEL-3192
+BABEL-3249
+BABEL-3121
+
+# JDBC bulk insert API seems to call SET FMTONLY ON without calling SET FMTONLY OFF, causing some spurious test failures.
+ignore#!#insertbulk
```
Comment (surendravishnoi) [inline test/JDBC/upgrade/13_4/schedule]: remove ignore files as well

**Review by surendravishnoi — COMMENTED** [review]
In current change, we are keeping verification_cleanup for target version in source version directory tagged with target version. Let us do other way around and keep verification_cleanup for target version in target directory tagged with source version.

File: .github/configuration/upgrade-test-configuration.yml (hunk 1 above)
Comment (rohit01010) [reply .github/configuration/upgrade-test-configuration.yml:10]: done

File: test/JDBC/upgrade/13_4/schedule (hunk 2 above)
Comment (rohit01010) [reply test/JDBC/upgrade/13_4/schedule]: done

File: test/JDBC/upgrade/13_4/schedule (hunk 3 above)
Comment (rohit01010) [reply test/JDBC/upgrade/13_4/schedule]: done

File: test/JDBC/upgrade/13_6/verification_cleanup/14_3/BABEL-3121-vu-verify.sql [hunk 4]
```diff
@@ -0,0 +1,44 @@
+USE db_babel_3121;
```
Comment (surendravishnoi) [inline test/JDBC/upgrade/13_6/verification_cleanup/14_3/BABEL-3121-vu-verify.sql]: why do we have 14_3 verification script inside 13_6 upgrade directory?

Comment (rohit01010) [reply test/JDBC/upgrade/13_6/verification_cleanup/14_3/BABEL-3121-vu-verify.sql]: removed

**Review by surendravishnoi — APPROVED** [review]

**Review by kuntalghosh — APPROVED** [review]

**Pull request merged**